File: mobile-app-distributor/PharmaFlowMobileBusiness/PharmaFlowMobileBusiness/Models/Item.swift
//
//  Item.swift
//  PharmaFlowMobileCustomer
//
//  Created by Dan on 2025-04-16.
//

import Foundation

struct Item: Identifiable, Codable {
    let id: UUID
    let codeName: String
    let refNumber: String?
    let gtin: String?
    let manufacturer: String
    let description: String?
    let officialUrl: String?
    let createdAt: Date?
    let canBeOrdered: Bool?
    let priceUsd: Decimal?

    enum CodingKeys: String, CodingKey {
        case id
        case codeName = "code_name"
        case refNumber = "ref_number"
        case gtin
        case manufacturer
        case description
        case officialUrl = "official_url"
        case createdAt = "created_at"
        case canBeOrdered = "can_be_ordered"
        case priceUsd = "price_usd"
    }
}

File: backend-app/PharmaFlowBackend/PharmaFlowBackend/Models/item.cs
using System;
using System.Collections.Generic;

namespace PharmaFlowBackend.Models;

public partial class item
{
    public Guid id { get; set; }

    public string code_name { get; set; } = null!;

    public string? ref_number { get; set; }

    public string? gtin { get; set; }

    public string manufacturer { get; set; } = null!;

    public string? description { get; set; }

    public string? official_url { get; set; }

    public DateTime? created_at { get; set; }

    public bool? can_be_ordered { get; set; }

    public decimal price_usd { get; set; }

    public virtual ICollection<lot> lots { get; set; } = new List<lot>();
}
